python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -118,10 +119,44 @@ def displayName(self):
     def processAlgorithm(self, parameters, context, feedback):
         expression = self.parameterAsString(parameters, self.EXPRESSION, context)
         layers = self.parameterAsLayerList(parameters, self.LAYERS, context)
-        layersDict = {}
+
         if layers:
+            layersDict = {}
```

Comment:
this change looks wrong ?

Reply:
As in, this layersDict = {} definitions should be moved out of the conditional statement?

Reply:
yes - it's referenced elsewhere so should always be made

Reply:
Fixed